test/index.ts

```diff
@@ -39,19 +37,21 @@ let options: { [key: string]: string | boolean | number } = {
 //
 // See https://github.com/mochajs/mocha/wiki/Using-mocha-programmatically#set-options for all available options
 
-for (let envVar of Object.keys(process.env)) {
+let environmentVariables = <{ [key: string]: string }>process.env;
+for (let envVar of Object.keys(environmentVariables)) {
     let match = envVar.match(/^mocha_(.+)/i);
     if (match) {
         let [, option] = match;
-        let value: string | number = process.env[envVar] || '';
-        if (!isNaN(parseInt(value))) {
-            value = parseInt(value);
+        let value: string | number = environmentVariables[envVar];
+        if (typeof value === 'string' && !isNaN(parseInt(value, undefined))) {
```

Comment:
Can process.env[envVar] be infinity? isNaN doesn't seem to cover infinity AFAIK.

Reply:
parseInt doesn't seem to be able to return Infinity